hstream-http-server/src/HStream/HTTP/Server/Query.hs

```diff
@@ -69,10 +69,22 @@ hstreamQueryToQueryBO (HSP.Query queryId (HSP.Info sqlStatement createdTime) (HS
 hstreamQueryNameIs :: T.Text -> HSP.Query -> Bool
 hstreamQueryNameIs name (HSP.Query queryId _ _) = (cbytesToText queryId) == name
 
--- removeQueryHandler :: HS.LDClient -> String -> Handler Bool
--- removeQueryHandler ldClient name = do
---   liftIO $ removeQuery ldClient (mkQueryName $ ZDC.pack name)
---   return True
+removeQueryHandler :: HS.LDClient -> Maybe ZK.ZHandle -> String -> Handler Bool
+removeQueryHandler ldClient zkHandle name = do
+  res <- liftIO $ do
+    queries <- HSP.withMaybeZHandle zkHandle HSP.getQueries
+    case find (hstreamQueryNameIs (T.pack name)) queries of
+      Just (HSP.Query queryId _ (HSP.Status status _)) -> do
+        if HSP.Terminated == status
+          then
+            do
+              _ <- forkIO $ HSP.withMaybeZHandle zkHandle $ HSP.removeQuery queryId
+              return True
+          else
+            -- TODO: should return detailed reason
+            return False
+      Nothing -> return False
```

Comment:
why find query and check the status when this is already been done by removeQuery?

Reply:
`HSP.withMaybeZHandle zkHandle $ HSP.removeQuery (T.pack name)` should be sufficient to remove a query, `removeQuery` can only remove a terminated query.